backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/floatingnotification/BpkFloatingNotification.kt

```diff
@@ -0,0 +1,180 @@
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.delay
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.icon.BpkIconSize
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkBorderRadius
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.utils.clickable
+
+@Composable
+fun BpkFloatingNotification(
+  hostState: BpkFloatingNotificationState,
+  modifier: Modifier = Modifier,
+) {
+  val data = hostState.data
+  val transitionDuration = 400
+  val fadeAnimationSpec: FiniteAnimationSpec<Float> = tween(durationMillis = transitionDuration)
+  val slideAnimationSpec: FiniteAnimationSpec<IntOffset> = tween(durationMillis = transitionDuration)
+  val requiredSize = if (LocalConfiguration.current.screenWidthDp < 360) {
+    RequiredSize(width = 288.dp, height = 52.dp)
+  } else if (LocalConfiguration.current.screenWidthDp in 360..512) {
+    RequiredSize(width = 312.dp, height = 52.dp)
+  } else {
+    RequiredSize(width = 400.dp, height = 72.dp)
+  }
+
+  AnimatedVisibility(
+    modifier = modifier,
+    visible = hostState.visible,
+    enter = slideInVertically(slideAnimationSpec, initialOffsetY = { it / 2 }) + fadeIn(animationSpec = fadeAnimationSpec),
+    exit = slideOutVertically(slideAnimationSpec, targetOffsetY = { it / 2 }) + fadeOut(animationSpec = fadeAnimationSpec)
+  ) {
+    Box(
+      modifier = Modifier
+        .fillMaxWidth()
+        .padding(start = BpkSpacing.Base, end = BpkSpacing.Base, bottom = 30.dp),
+      contentAlignment = Alignment.BottomCenter
+    ) {
+      Box(
+        modifier = Modifier
+          .clip(RoundedCornerShape(BpkBorderRadius.Md))
+          .background(color = BpkTheme.colors.corePrimary)
+          .requiredSize(height = requiredSize.height, width = requiredSize.width)
+          .padding(start = BpkSpacing.Base, end = BpkSpacing.Base),
+        contentAlignment = Alignment.CenterStart
+      ) {
+        Row(
+          horizontalArrangement = Arrangement.spacedBy(BpkSpacing.Base),
+          verticalAlignment = Alignment.CenterVertically
+        ) {
+          data?.icon?.let { icon ->
+            BpkIcon(
+              icon = icon,
+              contentDescription = null,
+              size = BpkIconSize.Small,
+              tint = BpkTheme.colors.textOnDark
+            )
+          }
+          BpkText(
+            modifier = Modifier.weight(0.8f),
+            text = data?.text ?: "",
```

Comment:
This feels a bit wrong - rather than checking if data is null for all composables here could we check that earlier to avoid these null checks here and make it a bit cleaner?

Reply:
moved the null-check to the beginning of method